tests/puppeteer/campaigns/functional/BO/catalog/brandsAndSuppliers/suppliers/01_CRUDSupplier.js

```diff
@@ -0,0 +1,133 @@
+const BrandsPage = require('@pages/BO/brands');
+const SuppliersPage = require('@pages/BO/suppliers');
+const AddSupplierPage = require('@pages/BO/addSupplier');
+const ViewSupplierPage = require('@pages/BO/viewSupplier');
+
+let browser;
+let page;
+let createSupplierData;
+let editSupplierData;
+
+// Init objects needed
+const init = async function () {
+  return {
+    boBasePage: new BOBasePage(page),
+    loginPage: new LoginPage(page),
+    dashboardPage: new DashboardPage(page),
+    brandsPage: new BrandsPage(page),
+    suppliersPage: new SuppliersPage(page),
+    addSupplierPage: new AddSupplierPage(page),
+    viewSupplierPage: new ViewSupplierPage(page),
+  };
+};
+
+// CRUD Supplier
+describe('Create, update and delete supplier', async () => {
+  // before and after functions
+  before(async function () {
+    browser = await helper.createBrowser();
+    page = await helper.newTab(browser);
+    this.pageObjects = await init();
+    createSupplierData = await (new SupplierFaker());
+    editSupplierData = await (new SupplierFaker());
+  });
+  after(async () => {
+    await helper.closeBrowser(browser);
+    await Promise.all([
+      files.deleteFile(createSupplierData.logo),
+      files.deleteFile(editSupplierData.logo),
+    ]);
+  });
+
+  // Login into BO and go to brands page
+  loginCommon.loginBO();
+
+  // Go to brands page
+  it('should go to brands page', async function () {
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.catalogParentLink,
+      this.pageObjects.boBasePage.brandsAndSuppliersLink,
+    );
+    await this.pageObjects.boBasePage.closeSfToolBar();
+    const pageTitle = await this.pageObjects.brandsPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.brandsPage.pageTitle);
+  });
+
+  // Go to suppliers page
+  it('should go to suppliers page', async function () {
+    await this.pageObjects.brandsPage.goToSubTabSuppliers();
+    const pageTitle = await this.pageObjects.suppliersPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.suppliersPage.pageTitle);
+  });
+  // 1: Create supplier
+  describe('Create supplier', async () => {
+    it('should go to new supplier page', async function () {
+      await this.pageObjects.suppliersPage.goToAddNewSupplierPage();
+      const pageTitle = await this.pageObjects.addSupplierPage.getPageTitle();
+      await expect(pageTitle).to.contains(this.pageObjects.addSupplierPage.pageTitle);
+    });
+
+    it('should create supplier', async function () {
+      const result = await this.pageObjects.addSupplierPage.createEditSupplier(createSupplierData);
+      await expect(result).to.equal(this.pageObjects.suppliersPage.successfulCreationMessage);
+    });
+  });
+  // 2: View supplier
+  describe('View supplier', async () => {
+    it('should view supplier', async function () {
+      await this.pageObjects.suppliersPage.viewSupplier(1);
```

Comment:
May be it's better to delete the variable in `viewSupplier()`

Reply:
comment added